internal/signerverifier/ssh/ssh.go

```diff
@@ -56,10 +60,14 @@ func (v *Verifier) Public() crypto.PublicKey {
 	return v.sshKey.(ssh.CryptoPublicKey).CryptoPublicKey()
 }
 
+func (v *Verifier) MetadataKey() *sv.SSLibKey {
```

Comment:
possibly a better way to do this? it's a bit wasteful to have both sshKey and sslibKey but I wonder if it's cheap enough to avoid the reverse conversion as well.

Reply:
I'd like to avoid this, because it implies that the two fields are in-sync, which is likely but not guaranteed.

IMO the most practical solution would be to have a struct that satisfies Key and Verifier interface, as in my original design, prior to https://github.com/gittuf/gittuf/pull/433. But that didn't fit in with the existing use of Key, Verifier, and Signer, and would thus require a more invasive patch.

Until then I recommend to ad-hoc create the SSlibKey from the contained sshKey . The code is already there, you could carve it out into a helper: 
https://github.com/gittuf/gittuf/blob/bcbb06557f29f4417a792877edb272b8a83a441c/internal/signerverifier/ssh/ssh.go#L107-L114 

(modulo passing the stored keyid, instead of regenerating it)

Reply:
Okay, I've used this in MetadataKey, but I've still retained that method for now.